pkg/cluster/manager/manager.go

```diff
@@ -134,3 +136,54 @@ func (m *Manager) sshTaskBuilder(name string, topo spec.Topology, user string, o
 		).
 		ClusterSSH(topo, user, opts.SSHTimeout, opts.SSHType, topo.BaseTopo().GlobalOptions.SSHType)
 }
+
+func (m *Manager) fillHostArch(s *cliutil.SSHConnectionProps, topo spec.Topology, gOpt *operator.Options, user string) error {
+	hostArch := map[string]string{}
+	var detectTasks []*task.StepDisplay
+	topo.IterInstance(func(inst spec.Instance) {
+		if _, ok := hostArch[inst.GetHost()]; ok {
+			return
+		}
+		hostArch[inst.GetHost()] = ""
+		if inst.Arch() != "" {
+			return
+		}
+
+		tf := task.NewBuilder().
+			RootSSH(
+				inst.GetHost(),
+				inst.GetSSHPort(),
+				user,
+				s.Password,
+				s.IdentityFile,
+				s.IdentityFilePassphrase,
+				gOpt.SSHTimeout,
+				gOpt.SSHType,
+				topo.BaseTopo().GlobalOptions.SSHType,
+			).
+			Shell(inst.GetHost(), "uname -m", "", false).
+			BuildAsStep(fmt.Sprintf("  - Detecting node %s", inst.GetHost()))
+		detectTasks = append(detectTasks, tf)
+	})
+	if len(detectTasks) == 0 {
+		return nil
+	}
+
+	ctx := ctxt.New(context.Background())
+	t := task.NewBuilder().
+		ParallelStep("+ Detect CPU Arch", false, detectTasks...).
+		Build()
+
+	if err := t.Execute(ctx); err != nil {
+		return perrs.Annotate(err, "failed to fetch cpu arch")
+	}
+
+	for host := range hostArch {
+		stdout, _, ok := ctxt.GetInner(ctx).GetOutputs(host)
```

Comment:
... and so the result handling should be just after the probe, as there might be multiple job for one host, it's possible to append a `Func` task after `Shell` to do so.

Reply:
What do you mean is make this task along with the later ones, such as check, deploy, scale-out? If so, IMO, it is best to have as little impact on the existing business as possible, and the CPU detection is a pre-action that needs to be done before the main logic(check, deploy, scale-out) is executed.  So, here I've implemented this action in an independent operation.

Reply:
Nope, I mean appending a `Func` task just after the `Shell` task that runs `uname`, but as we don't distinguish host by SSH port, it not necessary.